### PR TITLE
[NDB PERMISSION_DENIED error] Fix trusty stacktrace on ANDROID_EMULATOR

### PR DESCRIPTION
**Pull request by IvanBM18**
This PR comes as a need after the revert of this https://github.com/google/clusterfuzz/pull/5339 PR which introduced short circuit mechanisms to avoid DB queries, this however proved to provoke issues, so short circuit is no longer an option to fix any PERMISION_DENIED erros.

When we run jobs with an `ANDROID_EMULATOR` jobs, they all fail after executing a testcase. This happens because `ANDROID_EMULATOR` platform goes trough some legacy code(before the pre/main/pos separation of tasks) which expects the current bot to have access to DB or GCS.

This error only appear for `uworkers` and doesn't happen for `pre-emptible` bots. Because pre-emptibles do contain a trusted service account, where as uworkers dont.

## Changes
- When entering the process in which we analyze the stacktrace we now check if we can process this stacktrace or not, validating the kind of bot we are at.
- Creates Unit tests for this new edge cases

## Test performed
Changes initially present in dev since July 13th, and it didn't seem to non brake anything under ANDROID_EMULATOR jobs. Now for `ANDROID_EMULATOR` jobs.

Before:
image

After:
image

### CODE REVIEW

File: src/clusterfuzz/_internal/crash_analysis/stack_parsing/stack_symbolizer.py [hunk 1]
```diff
@@ -611,11 +595,22 @@ def _close_pipes(self):
       except ProcessLookupError:
         pass
 
+  def _can_process_trusty_stack_trace(self, trusty_app, trusty_bid) -> bool:
+    symbols_dir = environment.get_value('SYMBOLS_DIR')
+    return bool(
```
Comment (dylanjew) [inline src/clusterfuzz/_internal/crash_analysis/stack_parsing/stack_symbolizer.py:613]: can you add a comment similar to your description here to explain?

**Review by dylanjew — APPROVED** [review]

File: src/clusterfuzz/_internal/crash_analysis/stack_parsing/stack_symbolizer.py (hunk 1 above)
Comment (IvanBM18) [reply src/clusterfuzz/_internal/crash_analysis/stack_parsing/stack_symbolizer.py:613]: Done! i added them trough a pydoc instead so it looks cleaner :D

Comment (dylanjew) [reply src/clusterfuzz/_internal/crash_analysis/stack_parsing/stack_symbolizer.py:613]: thank you, I appreciate it!

**Review by dylanjew — APPROVED** [review]

**Review by javanlacerda — APPROVED** [review]
